lib/HLSL/HLOperationLower.cpp

```diff
@@ -7754,29 +7751,8 @@ void TranslateDefaultSubscript(CallInst *CI, HLOperationLowerHelper &helper,  HL
         if (RC == DXIL::ResourceClass::SRV) {
           // Invalid operations.
           Translated = false;
-          switch (IOP) {
-          case IntrinsicOp::IOP_InterlockedAdd:
-          case IntrinsicOp::IOP_InterlockedAnd:
-          case IntrinsicOp::IOP_InterlockedExchange:
-          case IntrinsicOp::IOP_InterlockedMax:
-          case IntrinsicOp::IOP_InterlockedMin:
-          case IntrinsicOp::IOP_InterlockedUMax:
-          case IntrinsicOp::IOP_InterlockedUMin:
-          case IntrinsicOp::IOP_InterlockedOr:
-          case IntrinsicOp::IOP_InterlockedXor:
-          case IntrinsicOp::IOP_InterlockedCompareStore:
-          case IntrinsicOp::IOP_InterlockedCompareExchange:
-          case IntrinsicOp::IOP_InterlockedCompareStoreFloatBitwise:
-          case IntrinsicOp::IOP_InterlockedCompareExchangeFloatBitwise: {
-            dxilutil::EmitErrorOnInstruction(
-                userCall, "Atomic operation targets must be groupshared or UAV.");
-            return;
-          } break;
-          default:
-            dxilutil::EmitErrorOnInstruction(userCall, "Invalid operation on typed buffer.");
-            return;
-            break;
-          }
+          dxilutil::EmitErrorOnInstruction(userCall, "Invalid operation on typed buffer.");
```

Comment:
Don't you mean on SRV?  I think this error already existed in the deleted block from which this was copied.

Reply:
I suppose so. As you say, this is a pre-existing error message that I've preserved. The only change was to generate the more specific error elsewhere. and avoid all this convolution here. That said, the only way I've managed to produce this error is through atomic ops.

Reply:
Actually, this is the one that I couldn't generate. I suspect it's here as a failsafe. It could probably be converted to an assert and/or internal error, though this gives us more information about where the problem is.

Reply:
maybe your const-ness fixes catch the SRV case earlier, preventing this from being hit.